utils/validate.ts

```diff
@@ -0,0 +1,28 @@
+interface ErrorInterface {
+  valid: boolean;
+  error: null | string;
+}
+
+export const invalidCharError: ErrorInterface = {
+  valid: false,
+  error: 'contains invalid characters'
+};
+export const validationSuccess = { valid: true, error: null };
+export const usernameTooShort = { valid: false, error: 'is too short' };
+export const usernameIsHttpStatusCode = {
+  valid: false,
+  error: 'is a reserved error code'
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+export const isNumeric = (num: any) => !isNaN(num);
+const validCharsRE = /^[a-zA-Z0-9\-_+]*$/;
+export const isHttpStatusCode = (str: string) =>
+  isNumeric(str) && parseInt(str, 10) >= 100 && parseInt(str, 10) <= 599;
```

Comment:
Since we know that the input is a string, we can make TS happy by reversing the order.  First `parseInt` since that accepts string arguments and then check if the output is both a number and in the range we're interested in.

Reply:
We still need to check if `str` is a number.  `parseInt` will provide either a number or `NaN`, we can check what it returns with `isNaN`.

Reply:
I've made those changes